back-nest/src/products/products.controller.ts

```diff
@@ -13,11 +15,43 @@ import { AuthGuard, RolesGuard } from '../authentication/auth.guard';
 
 @Controller('product')
 export class ProductController {
+  private readonly logger = new Logger(ProductController.name);
   constructor(private readonly productService: ProductService) {}
 
-  @Get()
-  async getAllProducts(): Promise<Product[]> {
-    return this.productService.getAllProducts();
+  @Get('')
+  async searchProducts(
+    @Query('searchTerm') title: string = '',
+    @Query('page') page: number = 1,
+    @Query('limit') limit: number = 20,
```

Comment:
create a searchProductsDTO query and use it only once. Also, you can't have numbers in query. They are always coming in string format. Consider transforming their types for further processing

```suggestion
    class SeachProductsDto {
      @IsString()
      title: string = ''
      
      @IsNumericString()
      page: string;
      
      @IsNumericString()
      limit: string
    }

    @Query() query: SearchProductsDto
```